odoo/addons/base/models/res_partner.py

```diff
@@ -162,6 +162,8 @@ def _split_street_with_params(self, street_raw, street_format):
     user_id = fields.Many2one('res.users', string='Salesperson',
       help='The internal user in charge of this contact.')
     vat = fields.Char(string='Tax ID', help="The Tax Identification Number. Complete it if the contact is subjected to government taxes. Used in some legal statements.")
+    same_vat = fields.Many2one('res.partner', string='Partner with same Tax ID', compute='_compute_same_vat', invisible=True)
```

Comment:
m2o field should be named with <model>_id 

I suggest same_tin_partner_id given than TIN (Tax Identification Number) is the default label. That field should also be store=False

Reply:
Aren't computed fields not stored by default?

Reply:
you make me doubt...

anyway, explicit is better than implicit ^^'

Reply:
From the doc

> computed fields are not stored by default, they are computed and returned when requested. Setting store=True will store them in the database and automatically enable searching

I will set it explicitly though

Reply:
And regarding the vat vs. tin thing... Isn't it more consistent to keep same_vat_partner_id as the field for the TIN is named vat ?

Reply:
haha ok you got a point :-) same_vat_partner_id is fine